include/qpid/dispatch/router.h

```diff
@@ -141,6 +141,11 @@ qd_router_forwarder_t *qd_router_get_forwarder(qd_address_treatment_t t);
 /** Retrieve the routers current memory usage (in bytes) */
 uint64_t qd_router_memory_usage(void);
 
+/** Return true if the router is running in test mode. Test mode can be enabled by passing the '-T' command line option
+ * to the router on startup.
+ */
+bool qd_router_test_mode_enabled(void);
```

Comment:
naming nit:

I like the old name better. The `qdr_core_test_hooks_enabled` suggests that there are some additional "hooks" that can be used to poke the router in tests.

The new name `qd_router_test_mode_enabled` suggests that the router itself somehow [works differently](https://jalopnik.com/understanding-the-test-mode-that-let-vw-trick-the-emiss-1732175835), that it has a test mode behavior and nontestmode behavior. And that is not true. Even the "crash me" config is actually a hook like the others.